src/cargo/core/package_id_spec.rs

```diff
@@ -551,11 +566,10 @@ mod tests {
         assert!(PackageIdSpec::parse("https://example.com#foo@1.2")
             .unwrap()
             .matches(foo));
-        // FIXME: The two tests below need to be corrected or adjusted - they should be passing.
-        assert!(!PackageIdSpec::parse("https://example.com/foo")
+        assert!(PackageIdSpec::parse("https://example.com/foo")
             .unwrap()
             .matches(foo));
-        assert!(!PackageIdSpec::parse("https://example.com/foo#1.2.3")
+        assert!(PackageIdSpec::parse("https://example.com/foo#1.2.3")
             .unwrap()
```

Comment:
What situation are you running into these package id specs existing that they should match the given package id?

From looking at the code, the implicit name is mostly to avoid redundancy, like if you depend on a package in the root of a git repo by the same name.

Reply:
In this issue:
https://github.com/hi-rustin/cargo-information/issues/57

```console
cargo info https://github.com/rust-lang/crates.io-index/clap
    Updating crates.io index
error: could not find `https://github.com/rust-lang/crates.io-index/clap` in registry `https://github.com/rust-lang/crates.io-index`
```

As you can see, it told me it could not find the clap crate from the crates.io.

I think this was a bug because the source ID doesn't have the package name in its URL. We shouldn't directly compare it.

Reply:
Where did you get that package id spec from?

```rust
$ cargo pkgid clap
https://github.com/rust-lang/crates.io-index#clap@4.4.10
```

Reply:
Oh. I copied it from here:
https://github.com/rust-lang/cargo/blob/a5fa676731c8e570bb9be9c50d2202acea84b6b6/src/cargo/core/package_id_spec.rs#L41

Reply:
Can I ask what is the difference between them? When to use the `"https://crates.io/foo"`?

Reply:
I wonder how representative that example is of real world spec ids...

Reply:
I will take a look at the details of the `pkgid` command. Thank you!